addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/protocol/WebSocketBase.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.samsungtv.internal.protocol;
+
+import java.io.IOException;
+import java.net.URI;
+
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.WebSocketAdapter;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Websocket base class
+ *
+ * @author Arjan Mels - Initial contribution
+ *
+ */
+class WebSocketBase extends WebSocketAdapter {
```

Comment:
Please use the WebsocketFactory of the core.net framework bundle. You acquire that as an OSGi service (via `@Reference`) in your factory or discovery or any other service class and pass it to where you need it.

Reply:
Will look into this; need to find some time for it.